### ISSUE TITLE
Implicit Clock used inconsistently

### ISSUE THREAD

**Issue by ghost** [jwt-scala/jwt-scala#270]
Hello. Thank you for your work in creating and maintaining this useful project.

I notice that some methods use an implicit Clock instance to allow easier unit testing. Unfortunately, the JwtCore decode methods do not currently have implicit argument lists, so decoding always uses the system clock. If an implicit Clock argument could be added to the JwtCore methods, the test Clock could be passed all the way down to JwtTime, which already supports it.

# Version
jwt-core 8.0.3

# Example Code
```scala
import java.time.Clock

import scala.util.{Failure, Success}

import pdi.jwt._
import pdi.jwt.exceptions.JwtExpirationException

object Example extends App {
  val startTime: Clock = MyTestClock(secondsSinceEpoch = 0)
  val claim = JwtClaim(args(0)).issuedNow(startTime).expiresIn(10)(startTime)
  val encoded = Jwt.encode(claim, "key", JwtAlgorithm.HS256)
  implicit val endTime: Clock = MyTestClock(secondsSinceEpoch = 5)
  Jwt.decode(encoded, "key", JwtAlgorithm.allHmac()) match {
    case Success(_) => Console.out.println("Expected")
    case Failure(_: JwtExpirationException) => Console.out.println("Actual")
    case _ =>
  }
}
```

# Expected output
"Expected"

# Actual output
"Actual"

**Comment by erwan** [comment]
thanks for the report, it looks like something we should fix indeed

**Comment by erwan** [comment]
Hey @axtsnee ,

I did a PR that I believe fixes your issue without changing the existing signature too much. In particular I wouldn't want to ask every single user to add an implicit clock to their code.

Now you can instantiate `Jwt()` instead of using the object to pass a custom Clock.

Would that work for you?

**Comment by ghost** [comment]
I think that is great! Thank you for solving this issue so quickly.

**Comment by erwan** [comment]
fixed in 9.0.0
